client/components/auth-form.jsx

```diff
@@ -0,0 +1,96 @@
+import React from 'react';
+import AppContext from '../lib/app-context';
+
+export default class AuthForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      username: '',
+      password: ''
+    };
+    this.signInButton = React.createRef();
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.handleUsernameChange = this.handleUsernameChange.bind(this);
+    this.handlePasswordChange = this.handlePasswordChange.bind(this);
+    this.handleSignUpClick = this.handleSignUpClick.bind(this);
+    this.handleSignInClick = this.handleSignInClick.bind(this);
+  }
+
+  handleSignUpClick(event) {
+    window.location.hash = 'sign-up';
+    this.context.route.path = 'sign-up';
```

Comment:
We never really _assign_ things to `state`, `props`, nor `context`, it's read-only. You shouldn't need to anyway as the app is supposed to parse the new hash and re-render when the hash changes